scrapi/harvesters/noaa_nodc.py

```diff
@@ -0,0 +1,226 @@
+                    './gmd:fileIdentifier',
+                    compose(lambda x: str(self.canonical_base_url).format(x), xml_text_only, single_result)
+                ),
+            },
+            'publisher': (
+                cite_stanza + 'gmd:citedResponsibleParty/gmd:CI_ResponsibleParty',
+                compose(extract_organization, single_result, filter_to_publishers),
+            ),
+            'providerUpdatedDateTime': ('./gmd:dateStamp/gco:DateTime/node()', compose(date_formatter, single_result)),
+            'languages': ('./gmd:language/gmd:LanguageCode', compose(language_codes, xml_text_only_list, coerce_to_list)),
+            'subjects': (id_stanza + 'gmd:descriptiveKeywords/gmd:MD_Keywords', lambda x: filter_keywords(x)),
+        }
+
+    search_base_url = 'http://data.nodc.noaa.gov/cgi-bin/oai-pmh?verb=ListRecords'
+    oai_ns = {
+        'oai_dc': 'http://www.openarchives.org/OAI/2.0/',
+        'dc': 'http://purl.org/dc/elements/1.1/',
+    }
+
+    def query_by_date(self, start_date, end_date):
+        '''Use OAI-PMH interface to get a list of dataset ids for the given date range'''
+        search_url_end = '&metadataPrefix=oai_dc&from=' + start_date + '&until=' + end_date
+        search_url = self.search_base_url + search_url_end
+
+        while True:
+            record_list = requests.get(search_url)
+            record_list_xml = etree.XML(record_list.content)
+            if record_list_xml.xpath('./oai_dc:error', namespaces=self.oai_ns):
+                break
+
+            for dataset in record_list_xml.xpath('./oai_dc:ListRecords/oai_dc:record', namespaces=self.oai_ns):
+                yield dataset.xpath('./oai_dc:header/oai_dc:identifier/node()', namespaces=self.oai_ns)[0]
+
+            token = record_list_xml.xpath('./oai_dc:ListRecords/oai_dc:resumptionToken/node()', namespaces=self.oai_ns)
+            if not token:
+                break
+
+            search_url = self.search_base_url + '&resumptionToken=' + token[0]
+
+    def harvest(self, start_date=None, end_date=None):
+        ''' First, get a list of all recently updated study urls,
+        then get the xml one by one and save it into a list
+        of docs including other information '''
+
+        start_date = (start_date or date.today() - timedelta(settings.DAYS_BACK)).isoformat()
+        end_date = (end_date or date.today()).isoformat()
+        start_date += 'T00:00:00Z'
+        end_date += 'T00:00:00Z'
+
+        # grab each of those urls for full content
+        xml_list = []
+        xml_base_url = self.canonical_base_url + '&view=xml'
+        for dataset_id in self.query_by_date(start_date, end_date):
+            try:
+                item_url = str(xml_base_url).format(dataset_id)
+                content = requests.get(item_url)
+            except exceptions.ConnectionError as e:
+                logger.info('Connection error: {}, wait a bit...'.format(e))
+                time.sleep(30)
+                continue
```

Comment:
continue will move on to the next dataset_id, skipping the current one. May not be what you meant to do.